components/concept/storage/src/main/java/mozilla/components/concept/storage/CreditCardsStorage.kt

```diff
@@ -0,0 +1,202 @@
+
+    /**
+     * Retrieves the credit card from the underlying storage layer by its unique identifier.
+     *
+     * @param guid Unique identifier for the desired credit card.
+     * @return [CreditCard] record.
+     */
+    suspend fun getCreditCard(guid: String): CreditCard
+
+    /**
+     * Retrieves a list of all the credit cards.
+     *
+     * @return A list of all [CreditCard].
+     */
+    suspend fun getAllCreditCards(): List<CreditCard>
+
+    /**
+     * Updates the fields in the provided credit card.
+     *
+     * @param creditCard A [CreditCard] record to update.
+     */
+    suspend fun updateCreditCard(creditCard: CreditCard)
+
+    /**
+     * Deletes the credit card with the given [guid].
+     *
+     * @return True if the deletion did anything, false otherwise.
+     */
+    suspend fun deleteCreditCard(guid: String): Boolean
+
+    /**
+     * Marks the credit card with the given [guid] as `in-use`.
+     *
+     * @param guid Unique identifier for the desired credit card.
+     */
+    suspend fun touchCreditCard(guid: String)
+
+    /**
+     * Inserts the provided address into the database, and returns
+     * the newly added [Address].
+     *
+     * @param addressFields A [NewAddressFields] record to add.
+     * @return [Address] for the added address.
+     */
+    suspend fun addAddress(addressFields: NewAddressFields): Address
+
+    /**
+     * Retrieves the address from the underlying storage layer by its unique identifier.
+     *
+     * @param guid Unique identifier for the desired address.
+     * @return [Address] record.
+     */
+    suspend fun getAddress(guid: String): Address
+
+    /**
+     * Retrieves a list of all the addresses.
+     *
+     * @return A list of all [Address].
+     */
+    suspend fun getAllAddresses(): List<Address>
+
+    /**
+     * Updates the fields in the provided address.
+     *
+     * @param address An [Address] record to update.
+     */
+    suspend fun updateAddress(address: Address)
+
+    /**
+     * Delete the address with the given [guid].
+     *
+     * @return True if the deletion did anything, false otherwise.
+     */
+    suspend fun deleteAddress(guid: String): Boolean
+
+    /**
+     * Marks the address with the given [guid] as `in-use`.
+     *
+     * @param guid Unique identifier for the desired address.
+     */
+    suspend fun touchAddress(guid: String)
+}
+
+/**
+ * Information about a credit card.
+ *
+ * @property guid The unique identifier for this credit card.
+ * @property ccName The credit card billing name.
+ * @property ccNumber The credit card number.
+ * @property ccExpMonth The credit card expiry month.
+ * @property ccExpYear The credit card expiry year.
+ * @property ccType The credit card network ID.
+ */
+data class CreditCard(
+    val guid: String,
+    val ccName: String,
+    val ccNumber: String,
+    val ccExpMonth: Long,
+    val ccExpYear: Long,
+    val ccType: String
+)
```

Comment:
There are also:
```

    i64 time_created;
    i64? time_last_used;
    i64 time_last_modified;
    i64 times_used;
```

Which could be useful for the detailed UX view of the credit cards.